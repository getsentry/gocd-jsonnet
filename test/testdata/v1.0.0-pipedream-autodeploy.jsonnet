local pipedream = import '../../v1.0.0/pipedream.libsonnet';

local pipedream_config = {
  name: 'example',
  auto_deploy: true,
  materials: {
    init_repo: {
      git: 'git@github.com:getsentry/init.git',
      shallow_clone: true,
      branch: 'master',
      destination: 'init',
    },
  },
};

local sample = {
  pipeline(region, config):: {
    region: region,
    config: config,
    materials: {
      example_repo: {
        git: 'git@github.com:getsentry/example.git',
        shallow_clone: true,
        branch: 'master',
        destination: 'example',
      },
    },
    stages: [
      {
        example_stage: {},
      },
    ],
  },
};

pipedream.render(pipedream_config, sample.pipeline)
