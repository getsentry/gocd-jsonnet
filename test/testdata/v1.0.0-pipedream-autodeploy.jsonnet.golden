{
   "example.yaml": {
      "format_version": 10,
      "pipelines": {
         "deploy-example": {
            "group": "example",
            "lock_behavior": "unlockWhenFinished",
            "materials": {
               "init_repo": {
                  "branch": "master",
                  "destination": "init",
                  "git": "git@github.com:getsentry/init.git",
                  "shallow_clone": true
               }
            },
            "stages": [
               {
                  "pipeline-complete": {
                     "approval": { },
                     "jobs": {
                        "start": {
                           "tasks": [
                              {
                                 "exec": {
                                    "command": true
                                 }
                              }
                           ]
                        }
                     }
                  }
               }
            ]
         }
      }
   }
}
